ssl/ssl.h

```diff
@@ -447,6 +447,15 @@ EXP_FUNC int STDCALL ssl_match_fingerprint(const SSL *ssl, const uint8_t* fp);
 EXP_FUNC int STDCALL ssl_match_spki_sha256(const SSL *ssl, const uint8_t* hash);
 
 /**
+ * @brief Check if SHA1 hash of CA public key matches the one given.
+ *
+ * @param ssl [in] An SSL object reference.
+ * @param fp [in] SHA1 hash to match against
+ * @return SSL_OK if the certificate is verified.
+ */
+EXP_FUNC int STDCALL ssl_match_auth_key_sha1(const SSL *ssl, const uint8_t* hash);
```

Comment:
>I was, wrongly, thinking that the authorityKeyIdentifier is integral part of the certificate, set in stone during the signing and it will always contain the SHA1 from the CA public key and cannot be faked.

No, it is indeed set in stone before the certificate is signed. But without having checking the signature using CA's public key, you don't know whether it was indeed signed by CA (and as such, authorityKeyInfo is validated by the CA), or it is signed by an adversary trying to send you a fake certificate.

If you want to do comparison based on SHA256 of the public key of the certificate, you can use the existing ssl_match_spki_sha256 — SPKI will not change when the certificate is reissued. But that way is still tied to a single certificate (as in, single website).